### DISCUSSION TITLE
Fundamentos do ASP.NET 6 - CRUD e EF - Necessidade de queries Sql usando EF

### DISCUSSION THREAD

**Discussion by lucastatagiba** [balta-io/forum#476] [Tira Dúvidas] [answered]
Olá, bom dia ! 
Duvida: 
Quando usamos o EF e temos contextos e tals, há possibilidade de fazer queries sql no código ? Se sim,  com contexto é realmente necessario, ja que ele faz o relacionamento sozinho com os mappings ? se sim, o Dapper é que tem o método Query no Sqlconnection, daria pra misturar Dapper com EF ? seria assim o jeito possivel de fazer as queries sql ?

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @lucastatagiba, como vai?

"Quando usamos o EF e temos contextos e tals, há possibilidade de fazer queries sql no código?"
- Sim, isto pode ser feito via Raw Queries como eu mostro no curso de Entity Framework:
https://balta.io/cursos/fundamentos-entity-framework/mapeando-queries-puras-e-views

"Se sim, com contexto é realmente necessario, ja que ele faz o relacionamento sozinho com os mapping?"
- Sim, é necessário o contexto, porém o mapeamento nem sempre, depende muito de como está seu banco. O EF sugere muita coisa, mas é "sugestão", eu recomendo mapear.
https://balta.io/cursos/fundamentos-entity-framework

" se sim, o Dapper é que tem o método Query no Sqlconnection, daria pra misturar Dapper com EF ?"
- Isto mesmo, o Dapper é uma extensão que tem este método e sim, dá pra misturar ele com EF

"seria assim o jeito possivel de fazer as queries sql?"
- É uma forma sim de fazer, mas depende do que você precisa...
Falei sobre isto neste vídeo: https://www.youtube.com/watch?v=sQAxDOeFFf0

**Reply by lucastatagiba** [reply]
Opa, boa tarde @andrebaltieri, vou bem e voce? 
Obrigado pela resposta, porém não achei a aula https://balta.io/cursos/fundamentos-entity-framework/mapeando-queries-puras-e-views, parece que não existe mais...

**Reply by andrebaltieri** [reply]
Bom dia, @lucastatagiba existe sim, acabei de abrir o link!

**Reply by lucastatagiba** [reply]
Verdade, desculpa por isso. Encontrei a aula e lembro de ter assistido e prestei atenção no dia, porém esqueci totalmente desse método, 🤦‍♂️. Obrigado @andrebaltieri !!
